File: moderate/Mth-To-Last-Element/MthToLastElement.cs
using System.IO;
using System.Collections.Generic;
using System;

class Program
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            // do something with line
            ShowMth(line);
        }
    }
    
    private static void ShowMth(String line){
		String[] elements = line.Split(' ');
		int leng = elements.Length-1;
		int number = Int32.Parse(elements[elements.Length-1]); 
		if (number <= elements.Length-1 && number!=0){
			Console.WriteLine(elements[leng-number]);
		}
		//for(String element : elements) System.out.println(element);
	}
}

File: moderate/Mth-To-Last-Element/MthToLastElement.java
import java.io.*;
public class Main {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            ShowMth(line);
        }
    }
    
    private static void ShowMth(String line){
		String[] elements = line.split(" ");
		int leng = elements.length-1;
		int number = Integer.valueOf(elements[leng]); 
		if (number <= elements.length-1 && number!=0)
		    System.out.println(elements[leng-number]);
	}
}
